### PR TITLE
Add web-nfc tests for default recordType/mediaType

### PR DESCRIPTION
**Pull request by xiuqijix**

### CODE REVIEW

File: web-nfc/NFCWriter_push.https.html [hunk 1]
```diff
@@ -377,4 +377,63 @@
     });
   });
 }, "NFCWriter.push should replace all previously configured push operations.");
-</script>
\ No newline at end of file
+
+nfc_test(async (t, mockNFC) => {
+  const writer = new NFCWriter();
+  await writer.push({ records: [{ data: test_text_data}] });
+  const message = createMessage([createTextRecord(test_text_data)]);
```
[the remark this answers is not in the document]
Comment (kaixinjxq) [reply web-nfc/NFCWriter_push.https.html]: Web NFC API:
```
typedef (DOMString or ArrayBuffer or NDEFMessageInit) NDEFMessageSource;

[SecureContext, Exposed=Window]
interface NFCWriter {
  constructor();
  Promise<void> push(NDEFMessageSource message, optional NFCPushOptions options={});
};
```

[nfc-helpers.js](https://github.com/web-platform-tests/wpt/blob/master/web-nfc/resources/nfc-helpers.js)(L126 ~ L129) handle the situation that NDEFMessageSource is DOMString or ArrayBuffer

BTW, the followings are the same reason.

[the remark this answers is not in the document]
Comment (kaixinjxq) [reply web-nfc/NFCWriter_push.https.html]: Done.

File: web-nfc/NFCWriter_push.https.html [hunk 2]
```diff
@@ -377,4 +377,63 @@
...
+  assertNDEFMessagesEqual(message, mockNFC.pushedMessage());
+}, "Test that recordType should be set to 'text' if NDEFRecordInit.record's \
+recordType is undefined and NDEFRecordInit.record's data is DOMString.");
+
+nfc_test(async (t, mockNFC) => {
+  const writer = new NFCWriter();
+  await writer.push({ records: [{ data: test_buffer_data}] });
+  const message = createMessage([createOpaqueRecord(test_buffer_data)]);
```
[the remark this answers is not in the document]
Comment (kaixinjxq) [reply web-nfc/NFCWriter_push.https.html]: Done.

File: web-nfc/NFCWriter_push.https.html [hunk 3]
```diff
@@ -377,4 +377,63 @@
...
+  assertNDEFMessagesEqual(message, mockNFC.pushedMessage());
+}, "Test that recordType should be set to 'json' if NDEFRecordInit.record's \
+recordType is undefined and NDEFRecordInit.record's data is not DOMString or \
+ArrayBuffer.");
+
+nfc_test(async (t, mockNFC) => {
+  const writer = new NFCWriter();
+  await writer.push({ records: [{ recordType: "text", data: test_text_data }] });
+  const message = createMessage([createTextRecord(test_text_data)]);
+  assertNDEFMessagesEqual(message, mockNFC.pushedMessage());
```
[the remark this answers is not in the document]
Comment (kaixinjxq) [reply web-nfc/NFCWriter_push.https.html]: Done.

File: web-nfc/NFCWriter_push.https.html [hunk 4]
```diff
@@ -377,4 +377,63 @@
...
+}, "Test that mediaType should be set to 'text/plain' if NDEFRecordInit.record's \
+recordType is 'text' and NDEFRecordInit.record's mediaType is undefined.");
+
+nfc_test(async (t, mockNFC) => {
+  const writer = new NFCWriter();
+  await writer.push({ records: [{ recordType: "opaque", data: test_buffer_data }] });
+  const message = createMessage([createOpaqueRecord(test_buffer_data)]);
```
[the remark this answers is not in the document]
Comment (kaixinjxq) [reply web-nfc/NFCWriter_push.https.html]: Done.

**Review by Honry — APPROVED** [review]
LGTM, thanks!

**Pull request merged**